clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1269,7 +1269,7 @@ public long position(TopicPartition partition) {
                 throw new IllegalArgumentException("You can only check the position for partitions assigned to this consumer.");
             Long offset = this.subscriptions.position(partition);
             if (offset == null) {
-                updateFetchPositions(Collections.singleton(partition));
+                updateFetchPositions(subscriptions.assignedPartitions());
```

Comment:
It is so we can call `consumer.seekToBeginning()` many times and then call `consumer.position(..)` many times without incurring a blocking round trip every time. 
The fetch positions are only updated if needed.